### PR TITLE
Add Elasticsearch version 5 and 6 support

### PR DESCRIPTION
**Pull request by mamedin**
- Added new elasticsearch17_download_url variable for ES version 1.7
- Added ESversionGt5 new set_fact variable for ES version greater than 5
- Updated README file: Included ES version 5 and 6 Thread Pools config
- Created new README-examples.md file with examples for Xenial, bionic
and CentOS and ES version 1,5 and 6.
- Removed ES_USER and ES_GROUP for ES version 6
https://www.elastic.co/guide/en/elasticsearch/reference/6.0/breaking_60_packaging_changes.html
- Removed path.conf variable for ES version 6
- Added new CentOS feature to install specific ES version

Connects to https://github.com/artefactual/archivematica/issues/1171

### CODE REVIEW

**Comment by scollazo** [comment]
Also you are using different code styles here, first you wrote:

```
- name: Download Elasticsearch deb (ES version >= 5.0.0)
  get_url: url={{ elasticsearch5_download_url }}/elasticsearch-{{ elasticsearch_version }}.deb dest=/tmp/elasticsearch-{{ elasticsearch_version }}.deb mode=0440
```
And later:
```
- name: "Add elasticsearch 5.x repo"
  yum_repository:
    name: "elasticsearch-5.x"
    description: "Elasticsearch repository for 5.x packages"
(...)
```

I think we should aim for the later, and if not updating all the instances of the previous style (that would be out of the scope of this pr), at least use the new one (each property on a new line) for the code we are adding.

**Comment by mamedin** [comment]
Thanks @scollazo , I changed the URL new variable for 1.7 version instead of version 5 or greater.

I created new 2 blocks to configure ES node with one line for each property.

It is ready for CR again.

**Comment by scollazo** [comment]
After our slack chat, what do you think about adding a safeguard that refuses to install es5/6 when 1.7 is installed?

File: tasks/RedHat.yml [hunk 1]
```diff
@@ -1,4 +1,18 @@
...
+- name: Get installed elasticsearch version
+  shell: rpm -qa --queryformat '%{version}' elasticsearch
+  args:
+    warn: false
+  register: installed_elasticsearch_version
+
+- name: Abort when elasticsearch installed version < 1.8 and version >=5 is going to be installed
+  fail:
+    msg: "Please uninstall elasticsearch {{ installed_elasticsearch_version.stdout }} before upgrading to {{ elasticsearch_version }} version"
+  when: installed_elasticsearch_version.stdout|trim != "" and installed_elasticsearch_version.stdout is version_compare('1.8', '<')  and elasticsearch_version is version_compare('5.0', '>=')
```
Comment (scollazo) [inline tasks/RedHat.yml]: I think that "and" is the default operator when you use more than one comparision, so writing it as:
```
when:
 - installed_elasticsearch_version.stdout|trim != "" 
 - installed_elasticsearch_version.stdout is version_compare('1.8', '<') 
 - elasticsearch_version is version_compare('5.0', '>=')
```

is easier to follow

File: tasks/RedHat.yml [hunk 2]
```diff
@@ -11,20 +25,40 @@
...
+
+- name: "Add elasticsearch 5.x repo"
+  yum_repository:
+    name: "elasticsearch-5.x"
+    description: "Elasticsearch repository for 5.x packages"
+    file: elasticsearch_repo
+    gpgkey: "https://artifacts.elastic.co/GPG-KEY-elasticsearch"
+    gpgcheck: yes
+    baseurl: "https://artifacts.elastic.co/packages/5.x/yum"
+  when: elasticsearch_version is version_compare('5.0', '>=') and elasticsearch_version is version_compare('6.0', '<')
```
Comment (scollazo) [inline tasks/RedHat.yml]: same than above, multiple "And" statements should go on a line on it's own to improve readability

**Comment by mamedin** [comment]
Thanks Santi, I changed these conditionals to multiline and they look better :)

Ready for CR again.

**Pull request merged**